tests/trainer/test_evaluation_loop.py

```diff
@@ -0,0 +1,42 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from unittest import mock
+
+from pytorch_lightning import Trainer
+from tests.helpers.boring_model import BoringModel
+
+
+@mock.patch("pytorch_lightning.trainer.evaluation_loop.EvaluationLoop.call_on_evaluation_epoch_end_hook")
```

Comment:
I don't think this test is required. this is already tested in `models/test_hooks` separately for val & test.

Reply:
I don't think there is a call count assertion test rather than just the order of the hooks.

Reply:
yes, there isn't right now, but I'd suggest updating this for all hooks there itself in a separate PR of course rather than updating it separately for only these 2 hooks.

Reply:
Doesn't sound like a good reason to skip valid tests if similar tests are not written for other hooks.